### DISCUSSION TITLE
Downloading large fastq files from google drive

### DISCUSSION THREAD

**Discussion by jdduprey** [sr320/course-fish546-2021#41] [Q&A] [answered]
The raw fastq files for my project are stored on google drive. I was wondering what the best way to download these from the command line is, similar to the examples we worked through for wget/curl and rsync?

**Comment by jdduprey** [comment, marked as answer]
I found a potential solution on stack exchange, will update if it works.

**Reply by jdduprey** [reply]
Okay the solution I found is this ugly block of code, and I'm unsure how I would scale it up to download the whole directory?

`fileId=13HVOweWXL5AAF2pIlEVMVn4suzPFrSQQ
fileName=Good-lib1_S1_L001_R1_001.fastq.gz
curl -sc /tmp/cookie "https://drive.google.com/uc?export=download&id=${fileId}" > /dev/null
code="$(awk '/_warning_/ {print $NF}' /tmp/cookie)"  
curl -Lb /tmp/cookie "https://drive.google.com/uc?export=download&confirm=${code}&id=${fileId}" -o ${fileName}`

This seems to work for a single file though.

**Reply by kubu4** [reply]
If you have the `fileID` and the `fileName` for all of your files, we can set up an array and loop through the array. However, I'm sure it's tedious to obtain both of those things.

Alternatively, I've used https://github.com/astrada/google-drive-ocamlfuse to mount my Google Drive on my computer and then transfer files that way. If I remember correctly, it's kind of slow, but it allows you to move things via the command line.

**Reply by kubu4** [reply]
>  we can set up an array and loop through the array

I can walk you through this.
